### PR TITLE
Multi select: keep selection after move

### LINKED ISSUE #19807: Moving multiple blocks deselects the blocks after a single action
Select multiple blocks and use the mover control to move up or down. 

Expected result: multiselection remains after moving.

Actual result: multiselection is cleared after moving.

GIF:

mover

CC: @ellatrix in case you have insights.

### PR DESCRIPTION
**Pull request by ellatrix**
## Description

Fixes #19807.

Since #19720, selection is cleared after moving a multi block selection. Unfortunately, there was no e2e test to check the behaviour. I'll add an e2e test in this PR.

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR. .

### CODE REVIEW

**Comment by jasmussen** [comment]
Fixes #19807

File: packages/block-editor/src/components/block-list/use-multi-selection.js [hunk 1]
```diff
@@ -147,7 +147,7 @@ export default function useMultiSelection( ref ) {
 		selectedBlockClientId,
 	] );
 
-	const onSelectionChange = useCallback( () => {
+	const onSelectionChange = useCallback( ( { isFinal } ) => {
```
Comment (aduth) [inline packages/block-editor/src/components/block-list/use-multi-selection.js]: Is `isFinal` meant to represent the end of a selection, corresponding to the `onSelectionEnd` event? Might it be something to name similar then, e.g. `isEnd` or `isSelectionEnd` ?

File: packages/block-editor/src/components/block-list/use-multi-selection.js [hunk 2]
```diff
@@ -159,6 +159,20 @@ export default function useMultiSelection( ref ) {
 
 		if ( startClientId.current === clientId ) {
 			selectBlock( clientId );
+
+			if ( isFinal ) {
```
Comment (aduth) [inline packages/block-editor/src/components/block-list/use-multi-selection.js]: So what exactly is this doing, and why does it fix? Is it something where the logic we've moved from `onSelectionEnd` is only intended to apply for single-block selections?

If so, I might suggest even some (useless, but clarifying) named variables, e.g.

```js
const isSingularSelection = startClientId.current === clientId;
if ( isSingularSelection ) {
```

And optionally some clarifying comments about how this particular logic is distinct for singular selections.

**Review by aduth — APPROVED** [review]

File: packages/e2e-tests/specs/editor/various/multi-block-selection.test.js [hunk 3]
```diff
@@ -408,4 +409,24 @@ describe( 'Multi-block selection', () => {
...
+		await testNativeSelection();
+		expect( await getSelectedFlatIndices() ).toEqual( [ 2, 3 ] );
+
+		await clickBlockToolbarButton( 'Move up' );
+
+		await testNativeSelection();
+		expect( await getSelectedFlatIndices() ).toEqual( [ 1, 2 ] );
+
+		expect( await getEditedPostContent() ).toMatchSnapshot();
+	} );
```
[the remark this answers is not in the document]
Comment (ellatrix) [reply packages/e2e-tests/specs/editor/various/multi-block-selection.test.js]: Thanks, I hadn't checked yet :D
I'll adjust the test. Looks like it only happens when selecting with the mouse

File: packages/block-editor/src/components/block-list/use-multi-selection.js (hunk 1 above)
Comment (ellatrix) [reply packages/block-editor/src/components/block-list/use-multi-selection.js]: Sounds good

**Pull request merged**